app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -713,4 +744,14 @@ public void showSimilarImageFragment(String originalFilePath, String possibleFil
         newFragment.setArguments(args);
         newFragment.show(getSupportFragmentManager(), "dialog");
     }
+
+    public void saveOnSubmit(){
+        flag++;
+        BasicKvStore storePrevTitleDesc = new BasicKvStore(this, "prevTitleDesc");
+        storePrevTitleDesc.putString("title", descriptionsAdapter.getTitle().toString());
+        for(int i = 0; i < descriptionsAdapter.getItemCount() - 1; i++) {
+            storePrevTitleDesc.putString("description"+Integer.toString(i), descriptionsAdapter.getDescriptions().get(i).getDescriptionText());
```

Comment:
Instead of storing `description0`, `description1` etc, store `description_<language_code>`.